rpc/transaction.go

```diff
@@ -46,8 +49,10 @@ func adaptTransaction(t TXN) (Transaction, error) {
 		json.Unmarshal(txMarshalled, &tx)
 		return tx, nil
 	default:
-		panic("not a transaction")
+		return nil, errors.New("Internal error with adaptTransaction() : unknown transaction type")
 	}
+	return nil, errors.New("Internal error with adaptTransaction()")
```

Comment:
this one became necessary due to missing default case in version switch, wdyt about adding more specific there i.e. "unknown transaction version %s" ?